packages/core/utils/src/dml/__tests__/entity-builder.spec.ts

```diff
@@ -1166,6 +1166,109 @@ describe("Entity builder", () => {
     })
   })
 
+  describe("Entity builder | primaryKey", () => {
+    test("should create both id fields and primaryKey fields", () => {
+      const model = new EntityBuilder()
+      const user = model.define("user", {
+        id: model.id({ primaryKey: false }),
```

Comment:
The main reason for my other suggestion was to eliminate this option entirely, so that it doesn't pollute the dml implementation. Fine to merge now, but let's see if we can get rid of it later on with an "inference" mechanism

Reply:
Did a dirty check for the inference, looks like we can get rid of this easily in the next PR.

Reply:
I like it! 💪